src/pages/Tracking/Tracking.jsx

```diff
@@ -110,24 +133,56 @@ class Tracking extends Component {
 	enableAllDeer = () => {
 		this.setState({ showAll: !this.state.showAll });
 
-		this.setState(({ deerStates }) => ({
-			deerStates: deerStates.map(deer => ({
-				...deer,
-				visible: !this.state.showAll,
-			})),
-		}));
+		this.setState(
+			({ deerStates }) => ({
+				deerStates: deerStates.map(deer => ({
+					...deer,
+					visible: !this.state.showAll,
+				})),
+			}),
+			() => this.filterButtonState()
+		);
 	};
 
+	filterButtonState(showAll = this.state.showAll) {
```

Comment:
is `showAll` ever passed in? If not this logic would be best inside the function, otherwise disregard